components/layout_2020/flexbox/layout.rs

```diff
@@ -264,10 +253,40 @@ impl FlexContainerConfig {
     ) -> bool {
         self.container_is_single_line &&
             item_with_auto_cross_size_stretches_to_line_size(
-                self.align_for(item_style.clone_align_self()),
+                AlignItems(self.resolve_align_self_for_child(item_style)),
                 item_margin,
             )
     }
+
+    fn resolve_reversable_flex_alignment(
+        &self,
+        align_flags: AlignFlags,
+        reversed: bool,
+    ) -> AlignFlags {
+        match (align_flags, reversed) {
+            (AlignFlags::FLEX_START, false) => AlignFlags::START | align_flags.flags(),
```

Comment:
Should this actually match `align_flags.value()` instead?
Otherwise `align_flags.flags()` is always 0 if `align_flags` is `FLEX_START` and `FLEX_END`.

Reply:
Yeah, I've switched this to match `align_flags.value()` after confirming that it doesn't work how we discussed it might.